pulp_2to3_migration/tests/functional/dynaconf_config.py

```diff
@@ -0,0 +1,3 @@
+from dynaconf import Dynaconf
+
+settings = Dynaconf(settings_files=['/etc/pulp/settings.py'])
```

Comment:
I just realized that this requires pulp :)
Any suggestion where to have credentials for the DB (for functional tests)?
hardcode in the code? set in travis as env vars? anything else?
The downside of setting env vars in travis is that it makes it inconvenient to run tests locally.

Reply:
You don't actually have to have Pulp. You can just use Dynaconf and a settings file.

Reply:
Maybe have a separate settings file tests?

Reply:
That's what I did here, but Travis doesn't have /etc/pulp/settings.py, because pulp is not there.
Are you suggesting to have a settings file for the func tests? (which is a form of hardcoding, and it needs to be changed if we changed defaults in pulp)

Reply:
disregard my comment, I didn't see your question when replied

Reply:
@goosemania If we don't want to put it in the same location, I suggest ```.config/pulp/settings.py```

Reply:
https://docs.pulpproject.org/installation/configuration.html#by-configuration-file

Reply:
Maybe this will work?

```diff
diff --git a/pulp_2to3_migration/tests/functional/dynaconf_config.py b/pulp_2to3_migration/tests/functional/dynaconf_config.py
index 8640a75..7653b14 100644
--- a/pulp_2to3_migration/tests/functional/dynaconf_config.py
+++ b/pulp_2to3_migration/tests/functional/dynaconf_config.py
@@ -1,3 +1,7 @@
 from dynaconf import Dynaconf
 
-settings = Dynaconf(settings_files=['/etc/pulp/settings.py'])
+try:
+    settings = Dynaconf(settings_files=['/etc/pulp/settings.py'])
+    settings.DATABASES
+except AttributeError:
+    settings = Dynaconf(settings_files=['~/.config/pulp/settings.py'])
```

Reply:
I think that would work.  I wonder if there's a more elegant way to do fallback from one possible location to another, maybe by listing both files in the proper order in the original invocation?

But I would be fine with merging this as-is.

Reply:
My understanding is that 2 files specified there are getting merged if both are present which might not be desirable.